laravel-app/app/Services/RegistrationService.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace App\Services;
+
+use App\Mail\EmailVerificationMail;
+use App\Models\User;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Mail;
+use Illuminate\Support\Str;
+
+class RegistrationService
+{
+    /**
+     * Logic for registering users
+     * @param array<string> $userData
+     */
+    public function registerUser(array $userData): void
+    {
+        $user = User::create([
+            'email' => $userData['email'],
+            'password' => Hash::make($userData['password']),
+            'username' => $userData['username'],
+            'email_verification_code' => Str::random(40),
+        ]);
+
+        $user->information()->create([
+            'first_name' => $userData['first_name'],
+            'last_name' => $userData['last_name'],
+        ]);
+
+        Mail::to($userData['email'])->send(new EmailVerificationMail($user));
+
+    }
+
+    /**
+     * Logic for verifying email
+     */
+    public function verifyEmail(string $verificationCode): bool
+    {
+        $user = User::where('email_verification_code', $verificationCode)->first();
+
+        if (!$user) {
+            return false;
+        }
+
+        if ($user->email_verified_at) {
+            return false;
+        }
+
```

Comment:
why separate if state when you have the same return result? Combine them instead!

Reply:
Yes sir 🙇